bittensor/_threadpool/priority_thread_pool_impl.py

```diff
@@ -166,11 +166,14 @@ def submit(self, fn, *args, **kwargs):
             start_time = time.time()
             if 'priority' in kwargs:
                 del kwargs['priority']
+            
+            timeout = kwargs.get('timeout', None)
+            if 'timeout' in kwargs:
```

Comment:
Can you add tests for this individually? There should be a test_priority_thread_pool.py